test/Hpack/ConfigSpec.hs

```diff
@@ -157,6 +158,14 @@ spec = do
         touch (dir </> "Setup.hs")
         getModules dir  "./." `shouldReturn` ["Foo"]
 
+  describe "getSignatures" $ around withTempDirectory $ do
+    it "returns signatures string in directory" $ \dir -> do
+      touch (dir </> "./Foo.hs")
+      touch (dir </> "./Bar/Baz.hs")
+      touch (dir </> "./Test.hsig")
+      touch (dir </> "./Test2.hsig")
+      getSignatures dir >>= (`shouldBe` Just (List ["Test", "Test2"]))
```

Comment:
This rejects hierarchical signatures, e.g. `Foo.Bar`.

Are hierarchical signatures allowed? cc @ezyang

Reply:
Yes, hierarchical signatures are allowed; the set of valid module names for signatures coincides with the set of valid module names.

Reply:
I understand. For example;

```
./Bar/Baz.hs
./Foo.hsig
./Foo/Bar.sig
```

Should be listed `Foo`, `Foo.Bar` in signatures?
I will be rewrite some code if correct.